File: chapt2/chapt2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self, node_name: str, name_value: str, age_value: int) -> None:
        print("PersonNode 构造函数被调用")
        super().__init__(node_name)
        self.name = name_value
        self.age = age_value

    def eat(self, food_name: str):
        """吃东西"""
        # print(f"{self.name}, {self.age}岁, 爱吃{food_name}")
        self.get_logger().info(f"{self.name}, {self.age}岁, 爱吃{food_name}")   

def main():
    rclpy.init()
    node = PersonNode("libai", "李白", 20)
    node.eat("月饼")
    rclpy.spin(node)
    rclpy.shutdown()

File: chapt2/chapt2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include <iostream>
#include <rclcpp/rclcpp.hpp>

using namespace std;

class PersonNode: public rclcpp::Node
{
private:
    string _name;
    int _age;

public:
    PersonNode(const string& node_name, const string& name, const int& age): Node(node_name)
    {
        this->_name = name;
        this->_age = age;
    }

    void eat(const string& food_name)
    {
        // RCLCPP_INFO(this->get_logger(), 
        //     "我是%s, %d岁, 爱吃%s", 
        //     this->_name.c_str(), 
        //     this->_age, 
        //     food_name.c_str());
        RCLCPP_INFO_STREAM(this->get_logger(),
            "我是" << this->_name << ", " << this->_age << "岁, 爱吃" << food_name);
    }
};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = make_shared<PersonNode>("person_node", "张三", 20);
    RCLCPP_INFO(node->get_logger(), "Hello, I am a person node.");
    node->eat("苹果");
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}